src/components/ClusterMap/cluster-map-view.js

```diff
@@ -69,37 +85,64 @@ export default class ClusterMapView extends React.Component {
       highlightedCats: [],
       highlightedLinks: [],
       highlightedProjects: [],
-      highlightedInfs: []
+      highlightedInfs: [],
+      hoverText: undefined,
+      uncertaintyHighlight: false
     });
   }
 
   highlightCat(category) {
-    this.setState({
-      highlightedCats: [category],
-      highlightedLinks: this.findLinksByCat(category).concat(
-        this.findProjectsByCat(category).map(p => this.findLinksByProject(p))
-      ),
-      highlightedProjects: this.findProjectsByCat(category),
-      highlightedInfs: this.findProjectsByCat(category).map(p =>
-        this.findInfsByProject(p)
-      )
-    });
+    if (this.props.selectedCat) {
+      this.setState({
+        highlightedCats: [this.props.selectedCat],
+        highlightedLinks: this.findLinksByCat(
+          this.props.categories.find(cat => cat.id === this.props.selectedCat)
+        ),
```

Comment:
in don't fully understand these changes, but there is some unexpected behaviour when highlighting. 

if a user picks a project and then continues to click through references in the details panel until there's another project, both are highlighted (one can continue with even more).
On reload, no highlights are recreated. 

This might be a hard one, so put it in a new ticket if it takes too long.

Reply:
the state is not ready yet, i will properly redo that whole thing, or maybe you have an idea